src/agent/debuglet.ts

```diff
@@ -420,20 +421,20 @@ export class Debuglet extends EventEmitter {
     }
   }
 
-  static getProjectIdFromMetadata(): Promise<string> {
-    return new Promise<string>((resolve, reject) => {
+  static getProjectIdFromMetadata(): Promise<string|undefined> {
+    return new Promise<string|undefined>((resolve, reject) => {
       metadata.project(
-          'project-id', (err: Error, res: any, projectId: string) => {
+          'project-id', (err: Error|null, res?: request.RequestResponse, projectId?: string) => {
             err ? reject(err) : resolve(projectId);
           });
     });
   }
 
-  static getClusterNameFromMetadata(): Promise<string> {
-    return new Promise<string>((resolve, reject) => {
+  static getClusterNameFromMetadata(): Promise<string|undefined> {
+    return new Promise<string|undefined>((resolve, reject) => {
       metadata.instance(
           'attributes/cluster-name',
-          (err: Error, res: any, clusterName: string) => {
+          (err: Error|null, res?: request.RequestResponse, clusterName?: string) => {
```

Comment:
question: would if work if you dropped the type annotations for parameters?

Reply:
Yes. Removed.